frontend/dashboard/pages/PageLayout/DashboardHeader/SmallHeaderMenu/SmallHeaderMenuItem/SmallHeaderMenuItem.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { type ReactElement } from 'react';
+import classes from './SmallHeaderMenuItem.module.css';
+import { NavLink, useLocation } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { DropdownMenu } from '@digdir/designsystemet-react';
+import type { NavigationMenuItem } from '../../../../../types/NavigationMenuItem';
+import { UrlUtils } from '@studio/pure-functions';
+
+export type SmallHeaderMenuItemProps = {
+  menuItem: NavigationMenuItem;
+  onClick: () => void;
+};
+
+export const SmallHeaderMenuItem = ({
+  menuItem,
+  onClick,
+}: SmallHeaderMenuItemProps): ReactElement => {
+  const { t } = useTranslation();
+  const location = useLocation();
+  const currentRoutePath: string = UrlUtils.extractSecondLastRouterParam(location.pathname);
+
+  if (menuItem.action.type === 'button') {
+    const handleClick = () => {
+      onClick();
+      menuItem.action.type === 'button' && menuItem.action.onClick();
```

Comment:
Is this check not redundant because of the if statement on line 22?
```suggestion
      menuItem.action.onClick();
```

Reply:
I've tried to do that, but for some reason i get: 
```
Property 'onClick' does not exist on type 'StudioProfileMenuItemAction<StudioProfileMenuItemType>'.
  Property 'onClick' does not exist on type 'StudioProfileMenuItemLinkAction'.
  ```

Reply:
Oh ok, then it's fine